src/analysis/normed_space/real_inner_product.lean

```diff
@@ -221,6 +222,48 @@ instance inner_product_space_is_normed_space : normed_space ℝ α :=
 
 end norm
 
+section instances
+/-- The standard Euclidean space, `fin n → ℝ`. -/
+def euclidean_space (n : ℕ) : Type := fin n → ℝ
+
+local attribute [reducible] euclidean_space
```

Comment:
Why is this required?

Reply:
Without it, Lean can't see through the definition of `euclidean_space` to find the instances on which `inner_product_space` depends.

```
234:1: failed to synthesize type class instance for
n : ℕ
⊢ add_comm_group (euclidean_space n)
```

I don't know how the merits of this approach (taken from what's done for the manifold version) compare to other approaches for providing those instances.

Reply:
It's really just in this section do register the instances. The other option would be to `unfold` it in the instance proofs.